fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -695,7 +697,12 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         startupPathProvider.onIntentReceived(intent)
     }
 
-    open fun handleNewIntent(intent: Intent) {
+    @VisibleForTesting
+    internal fun handleNewIntent(intent: Intent) {
+        if (this is ExternalAppBrowserActivity) {
+            return
+        }
```

Comment:
🚗 it feels a little code smelly to me for an `open` type to have knowledge of its subclasses. maybe `if (this !is HomeActivity)` could work instead?

Reply:
Agreed, but it is temporary. Wouldn't this always be false though because of type inheritance, since an `ExternalAppBrowserActivity`'s super type is `HomeActivity`?

Reply:
> Wouldn't this always be false though because of type inheritance, since an ExternalAppBrowserActivity's super type is HomeActivity

Yep, this. Agree it's not ideal but as this gets further decoupled we'll be able to refine these checks

Reply:
in fact this check will just be changed to check a boolean if `ExternalAppBroswerActivity` is no more

Reply:
Do you mean that we'll be using a single activity and deliberately tracking whether we are in custom tab mode, or something like that?

Reply:
Yeah exactly that - essentially using a flag in the `Activity` to track if is 'external' or not